pkg/kubevirt/core/util.go

```diff
@@ -163,26 +163,47 @@ ethernets:
 	return interfaces, networks, networkData
 }
 
+func findDiskByVolumeName(name string, disks []kubevirtv1.Disk) *kubevirtv1.Disk {
+	for _, disk := range disks {
+		if name == disk.Name {
+			return &disk
+		}
+	}
+	return nil
+}
+
+func generateDefaultDisk(name string) kubevirtv1.Disk {
+	return kubevirtv1.Disk{
+		Name: name,
+		DiskDevice: kubevirtv1.DiskDevice{
+			Disk: &kubevirtv1.DiskTarget{
+				Bus: "virtio",
+			},
+		},
+	}
+}
+
 func buildVolumes(
 	machineName, namespace, userDataSecretName, networkData string,
 	rootVolume cdicorev1alpha1.DataVolumeSpec,
 	additionalVolumes []api.AdditionalVolumeSpec,
+	providedDisks []kubevirtv1.Disk,
```

Comment:
I would use `configuredDisks` as it's a bit more concrete.
```suggestion
	configuredDisks []kubevirtv1.Disk,
```